Sources/NIO/SocketChannel.swift

```diff
@@ -482,8 +482,11 @@ class BaseSocketChannel<T: BaseSocket>: SelectableChannel, ChannelCore {
             pipeline.fireErrorCaught0(error: err)
         }
 
-        executeAndComplete(promise) {
+        do {
             try socket.close()
+        } catch {
+            promise?.fail(error: error)
+            return
```

Comment:
Wait, why are we returning here? If we *fail* to close the socket we don't cancel the pending writes?

Reply:
maybe we should still do ... Good question. I guess there is not much we can do in this case.